misc/multiaddr/src/onion_addr.rs

```diff
@@ -0,0 +1,54 @@
+use std::borrow::Cow;
+use std::fmt::Debug;
+use serde::export::Formatter;
+use std::fmt;
+
+/// Represents an Onion v3 address
+#[derive(Clone)]
+pub struct Onion3Addr<'a>(Cow<'a, [u8; 35]>, u16);
+
+impl<'a> Onion3Addr<'a> {
+    /// Return the hash of the public key as bytes
+    pub fn hash(&self) -> &[u8; 35] {
+        self.0.as_ref()
+    }
+
+    /// Return the port
+    pub fn port(&self) -> u16 {
+        self.1
+    }
+
+    /// Consume this instance and create an owned version containing the same address
+    pub fn into_owned<'b>(self) -> Onion3Addr<'b> {
```

Comment:
To align this with `Protocol`, could you change the name into `acquire`? `Onion3Addr::into_owned` is the better name but we can not change `Protocol::acquire` or we would break backwards compatibility and I think it would be nice to have consistent naming in this crate.

Reply:
Sure thing 👍